flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/LookupJoinTestPrograms.java

```diff
@@ -0,0 +1,287 @@
+                                            "+I[6, 7.65, null, null, null, null, null, null]",
+                                            "+I[5, 12.78, 2, Alice, 32, San Francisco, California, 95016]")
+                                    .consumedAfterRestore(
+                                            "+I[7, 17.58, 6, Joana, 54, Atlanta, Georgia, 30033]",
+                                            "+I[9, 143.21, null, null, null, null, null, null]")
+                                    .build())
+                    .runSql(
+                            "INSERT INTO sink_t SELECT "
+                                    + "O.order_id, "
+                                    + "O.total, "
+                                    + "C.id, "
+                                    + "C.name, "
+                                    + "C.age, "
+                                    + "C.city, "
+                                    + "C.state, "
+                                    + "C.zipcode "
+                                    + "FROM orders_t as O "
+                                    + "LEFT JOIN customers_t FOR SYSTEM_TIME AS OF O.proc_time AS C "
+                                    + "ON O.customer_id = C.id AND C.age > 30 AND O.customer_id <> 3")
+                    .build();
+
+    static final TableTestProgram LOOKUP_JOIN_ASYNC_HINT =
+            TableTestProgram.of("lookup-join-async-hint", "validates lookup join with async hint")
+                    .setupTableSource(CUSTOMERS)
+                    .setupTableSource(ORDERS)
+                    .setupTableSink(
+                            SinkTestStep.newBuilder("sink_t")
+                                    .addSchema(SINK_SCHEMA)
+                                    .consumedBeforeRestore(
+                                            "+I[1, 44.44, 3, Claire, 37, Austin, Texas, 73301]",
+                                            "+I[2, 100.02, 5, Jake, 42, New York City, New York, 10001]",
+                                            "+I[4, 92.61, 2, Alice, 32, San Francisco, California, 95016]",
+                                            "+I[3, 23.89, 1, Bob, 28, Mountain View, California, 94043]",
+                                            "+I[6, 7.65, 4, Shannon, 29, Boise, Idaho, 83701]",
+                                            "+I[5, 12.78, 2, Alice, 32, San Francisco, California, 95016]")
+                                    .consumedAfterRestore(
+                                            "+I[7, 17.58, 6, Joana, 54, Atlanta, Georgia, 30033]",
+                                            "+I[9, 143.21, 1, Bob, 28, San Jose, California, 94089]")
+                                    .build())
+                    .runSql(
+                            "INSERT INTO sink_t SELECT "
+                                    + "/*+ LOOKUP('table'='C', 'async'='true', 'output-mode'='allow_unordered') */ "
```

Comment:
Do you know if those options are stored in the CompiledPlan? I looked shortly into the `StreamExecLookupJoin` and I believe e.g. `async` should be serialized in `asyncOptions`, but I can't find such a field in any of the plans (even the old ones).

Reply:
Fixed this. Now the updated plan has the async options.